src/appservice/Appservice.ts

```diff
@@ -508,6 +517,23 @@ export class Appservice extends EventEmitter {
         }
     }
 
+    /**
+     * Adds a preprocessor to the event pipeline. When this appservice encounters an ephemeral event, it
+     * will try to run it through the preprocessors it can in the order they were added.
+     * @param {IPreprocessor} preprocessor the preprocessor to add
+     */
+    public addEphemeralEventPreprocessor(preprocessor: IPreprocessor): void {
```

Comment:
In practice I think we can support the pre-existing pre-processors: people have to opt in to this.